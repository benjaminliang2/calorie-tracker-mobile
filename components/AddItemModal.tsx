import { useRef } from "react"
import { useState, useEffect } from "react"
import { Button, ImageBackground, Modal, Pressable, StyleSheet, Text, TextInput, TouchableOpacity, View } from "react-native"
import * as ImagePicker from 'expo-image-picker'
import { useAppDispatch } from "../redux/hooks"
import Icon from 'react-native-vector-icons/FontAwesome5'
import CheckBox from 'expo-checkbox'

import { addItem } from "../redux/features/NutritionSlice"

interface Props {
    showModal: boolean, 
    setShowModal: Function
}

export const AddItemModal = ({ showModal, setShowModal }) => {
    let today = new Date()
    const dispatch = useAppDispatch()
    const [item, setItem] = useState({
        image: 'https://reactnative.dev/img/tiny_logo.png',
        name: '',
        id: '',
        calories: 0,
        proteins: 0,
        carbohydrates: 0,
        fats: 0,
    })
    const calorieInput = useRef(null)
    const carbInput = useRef(null)
    const proteinInput = useRef(null)
    const fatInput = useRef(null)

    useEffect(()=>{
        if (item.id !== ''){
            dispatch(addItem(item))
            setShowModal(false)
        }
    },[item.id])
    // const pickImage = async () => {
    //     // No permissions request is necessary for launching the image library
    //     let result = await ImagePicker.launchImageLibraryAsync({
    //         allowsEditing: true,
    //         aspect: [3, 3],
    //         quality: 1,
    //     });

    //     console.log(result);

    //     if (!result.cancelled) {
    //         setItem(prevState => ({ ...prevState, image: result.uri }))

    //     }
    // };

    const takePicture = async () => {
        let result = await ImagePicker.launchCameraAsync({
            allowsEditing: true,
            aspect: [3, 3],
            quality: 0.5,
        })
        if (!result.cancelled) {
            const {uri} = result as ImagePicker.ImageInfo
            setItem(prevState => ({ ...prevState, image: uri }))
        }
    }

    return (<>
        <View style={styles.centeredView}>
            <Modal
                animationType="slide"
                transparent={true}
                visible={showModal}
                onRequestClose={() => {
                    alert('Modal has been closed.');
                    setShowModal(!showModal);
                }}
            >
                <View style={styles.centeredView}>
                    <View style={styles.modalView}>

                        <Pressable onPress={takePicture}>
                            <ImageBackground
                                style={styles.image}
                                resizeMode='cover'
                                source={{ uri: item.image }}
                            >
                                <Icon name='camera' size={25} color="#65C18C" style={{ margin: 10, position: 'absolute', right: 0, bottom: 0, }} />
                            </ImageBackground>
                        </Pressable>


                        <Text>Enter Item</Text>

                        <TextInput
                            style={styles.input}
                            onChangeText={(event) => setItem(prevState => ({ ...prevState, name: event }))}
                            value={item.name}
                        />

                        <View style={{ flexDirection: 'row' }}>


                            <View style={styles.macroColumn}>
                                <Pressable style={styles.container} onPress={() => calorieInput?.current?.focus()}>
                                    <Text style={styles.macro}> Calories</Text>
                                    <TextInput
                                        ref={calorieInput}
                                        style={styles.input}
                                        keyboardType={'numeric'}
                                        returnKeyType='done'
                                        onChangeText={(event) => setItem(prevState => ({ ...prevState, calories: parseInt(event) }))}
                                        value={`${item.calories}`}
                                        placeholder='0'
                                    />
                                    <Icon name='edit' size={18} color="#65C18C" style={{ margin: 10, position: 'absolute', right: 0, bottom: 0, }} />

                                </Pressable>

                                <Pressable style={styles.container} onPress={() => proteinInput?.current?.focus()}>
                                    <Text style={styles.macro}> Proteins</Text>
                                    <TextInput
                                        ref={proteinInput}
                                        style={styles.input}
                                        keyboardType={'numeric'}
                                        returnKeyType='done'
                                        onChangeText={(event) => setItem(prevState => ({ ...prevState, proteins: parseInt(event) }))}
                                        value={`${item.proteins}`}
                                        placeholder='0'
                                    />
                                    <Icon name='edit' size={18} color="#65C18C" style={{ margin: 10, position: 'absolute', right: 0, bottom: 0, }} />
                                </Pressable>
                            </View>


                            <View style={styles.macroColumn}>
                                <Pressable style={styles.container} onPress={() => carbInput?.current?.focus()}>
                                    <Text style={styles.macro}> Carbs</Text>
                                    <TextInput
                                        ref={carbInput}
                                        style={styles.input}
                                        keyboardType={'numeric'}
                                        returnKeyType='done'
                                        onChangeText={(event) => setItem(prevState => ({ ...prevState, carbohydrates: parseInt(event) }))}
                                        value={`${item.carbohydrates}`}
                                        placeholder='0'
                                    />
                                    <Icon name='edit' size={18} color="#65C18C" style={{ margin: 10, position: 'absolute', right: 0, bottom: 0, }} />
                                </Pressable>
                                <Pressable style={styles.container} onPress={() => fatInput?.current?.focus()}>
                                    <Text style={styles.macro}> Fats</Text>
                                    <TextInput
                                        ref={fatInput}
                                        style={styles.input}
                                        keyboardType={'numeric'}
                                        returnKeyType='done'
                                        onChangeText={(event) => setItem(prevState => ({ ...prevState, fats: parseInt(event) }))}
                                        value={`${item.fats}`}
                                        placeholder='0'
                                    />
                                    <Icon name='edit' size={18} color="#65C18C" style={{ margin: 10, position: 'absolute', right: 0, bottom: 0, }} />
                                </Pressable>
                            </View>


                        </View>
                        <View style={{ flexDirection: 'row', justifyContent: 'center', alignItems: 'center', }}>
                            <CheckBox
                                style={{ margin: 10 }}
                                value={false}
                                onValueChange={() => console.log("checkbox")}
                                color={'#4630EB'}
                            />
                            <Text>Add to Favorites</Text>
                        </View>
                        <View style={{ flexDirection: 'row' }}>
                            <Pressable style={styles.button} onPress={() => setShowModal(false)}>
                                <Text>Cancel</Text>
                            </Pressable>
                            <Pressable
                                onPress={() => item.name.trim() ?                               
                                    setItem(prevState => ({ ...prevState, id: item.name + today.getTime() }))  
                                    :
                                    alert('Please enter name')
                                }
                                style={styles.button}
                            >
                                <Text>Add</Text>
                            </Pressable>

                        </View>

                    </View>
                </View>
            </Modal>
        </View>
    </>)
}

const styles = StyleSheet.create({
    centeredView: {
        flex: 1,
        // justifyContent: 'center',
        // paddingTop: 22,
        marginTop: 50,
        backgroundColor: '#A4BE7B',
    },
    modalView: {
        // margin: 20,
        borderRadius: 20,
        padding: 35,
        // alignItems: 'center',
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.35,
        shadowRadius: 8,
        elevation: 5,
    },

    image: {
        height: 150,
        width: 150,
        position: 'relative',
        alignSelf: 'center',

    },
    macroColumn: {
        flex: 1
    },
    container: {
        backgroundColor: 'white',
        alignItems: 'center',
        height: 100,
        margin: 10,
        borderRadius: 20,
        justifyContent: 'space-evenly',
        position: 'relative',


    },
    macro: {
        fontFamily: 'MontserratSemiBold',
        fontSize: 20,
        color: '#B2B2B2'

    },
    input: {
        backgroundColor: "#fff",
        fontSize: 30,
        // width: "50%",
        textAlign: 'center',
    },
    button: {
        alignItems: "center",
        backgroundColor: "#fff",
        margin: 10,
        padding: 10,
        flex: 1,
    },
})
