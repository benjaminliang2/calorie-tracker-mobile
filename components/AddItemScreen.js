import { useState, useEffect } from "react"
import { StyleSheet, Text, TextInput, TouchableOpacity, View } from "react-native"
import { useDispatch } from "react-redux"

import { addItem } from "../redux/features/TodaysNutritionSlice"
import { Background1 } from "./gradients/Background1"


export const AddItemScreen = () => {
    const dispatch = useDispatch()
    const [item, setItem] = useState({
        name: '',
        calories: 0,
        proteins: 0,
        carbohydrates: 0,
        fast: 0,
    })

    return (<>
        <Background1>
            <View >
                <Text>Enter Item</Text>

                <TextInput
                    style={styles.input}
                    onChangeText={(event) => setItem(prevState => ({...prevState, name: event})) }
                    value={item.name}
                />

                <View style={{ flexDirection: 'row' }}>
                    <View style={styles.row2}>
                        <Text> Calories</Text>
                        <TextInput
                            style={styles.input}
                            onChangeText={(event) => setItem(prevState => ({...prevState, calories: parseInt(event)})) }
                            keyboardType={'numeric'}
                            value={item.calories}
                        />
                        <Text> Protein</Text>
                        <TextInput
                            style={styles.input}
                            onChangeText={(event) => setItem(prevState => ({...prevState, proteins: parseInt(event)})) }
                            keyboardType={'numeric'}
                            value={item.proteins}
                        />
                    </View>
                    <View style={styles.row2}>
                        <Text> Carbs</Text>
                        <TextInput
                            style={styles.input}
                            onChangeText={(event) => setItem(prevState => ({...prevState, carbohydrates: parseInt(event)})) }
                            keyboardType={'numeric'}
                            value={item.carbohydrates}
                        />
                        <Text> Fat</Text>
                        <TextInput
                            style={styles.input}
                            onChangeText={(event) => setItem(prevState => ({...prevState, fats: parseInt(event)})) }
                            keyboardType={'numeric'}
                            value={item.fats}
                        />
                    </View>
                </View>

                <View>
                    <TouchableOpacity
                        style={styles.button}
                        onPress={() => {
                            dispatch(addItem(item))
                            console.log(item)
                        }}
                    >
                        <Text>Add</Text>
                    </TouchableOpacity>
                </View>

            </View>
        </Background1>
    </>)
}

const styles = StyleSheet.create({
    row2: {
        flex: 2
    },
    input: {
        backgroundColor: "#fff",
        width: '50%'
    },
    button: {
        alignItems: "center",
        backgroundColor: "#DDDDDD",
        padding: 10,
    },
})
