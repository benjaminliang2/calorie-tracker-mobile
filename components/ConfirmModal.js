import { Button, Modal, Pressable, StyleSheet, Text, View } from "react-native";
import { useDispatch } from "react-redux"


export const ConfirmModal = ({ title, action, actionName, value, setShowModal, showModal }) => {
    const dispatch = useDispatch()

    return <>
        {/* <View style={styles.centeredView}> */}
        <Modal
            animationType="slide"
            transparent={true}
            visible={showModal}
            onRequestClose={() => {
                Alert.alert('Modal has been closed.');
                setShowModal(!showModal);
            }}
        >
            <View style={styles.centeredView}>
                <View style={styles.modalView}>
                    <Text>{title}</Text>
                    <View style={styles.buttonRow}>
                        <Button title="Cancel" onPress={() => setShowModal(false)} style={styles.cancelButton} />
                        <Pressable style={styles.actionButton} onPress={() => (dispatch(action(value)), setShowModal(false))}>
                            <Text style={styles.actionText}>{actionName}</Text>
                        </Pressable>
                    </View>
                </View>
            </View>
        </Modal>
        {/* </View> */}
    </>
}

const styles = StyleSheet.create({
    centeredView: {
        flex: 1,
        justifyContent: 'center',
        // alignItems: 'center',
    },
    modalView: {
        // margin: 20,
        backgroundColor: '#fff',
        borderRadius: 20,
        padding: 35,
        alignItems: 'center',
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.35,
        shadowRadius: 8,
        elevation: 5,
    },
    buttonRow: {
        flexGrow: 1,
        flexDirection: 'row',
        justifyContent: 'flex-end',

    },
    cancelButton: {
        flex: 1,

    },
    actionButton: {
        backgroundColor: 'red',
        alignItems: 'center',
        justifyContent: 'center',
        paddingVertical: 12,
        paddingHorizontal: 26,
        borderRadius: 4,
        // flex: 1,

    },
    actionText:{
        fontSize: 16,
        fontWeight: 'bold',
        color: 'white',
        
    }

})