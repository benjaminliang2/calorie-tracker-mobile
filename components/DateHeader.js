import { Text, View } from "react-native"

import Icon from 'react-native-vector-icons/FontAwesome5'

export const DateHeader = () => {

    return (<>
        <View style={{ flexDirection: 'row', justifyContent: 'space-between', width: '100%', alignItems: 'center', }}>
            <View style={{margin: 10}}>
                <Text style={{ fontFamily: 'MontserratRegular', fontSize: 14, lineHeight: 17, }} >Today</Text>
                <Text style={{ fontFamily: 'MontserratBold', fontSize: 18, lineHeight: 29, }}>Sat, Nov 12</Text>
            </View>
            <Icon name='calendar-alt' size={30} color="black"  style={{margin: 10}}/>

        </View>
    </>)
}