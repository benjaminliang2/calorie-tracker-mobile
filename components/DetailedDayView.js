import { useSelector } from "react-redux"
import { SafeAreaView, ScrollView, Text, TouchableOpacity, View } from "react-native"



export const DetailedDayView = ({navigation}) => {
    const { items } = useSelector((store) => store.todaysNutrition)

    return (<>
        <SafeAreaView>
            {/* <TouchableOpacity onPress={()=> navigation.goBack()}>
                <Text> Back </Text>
            </TouchableOpacity> */}
            <ScrollView>
                {items.map(item => (
                    <FoodCard item={item} />
                ))}
            </ScrollView>

        </SafeAreaView>
    </>)



}

export const FoodCard = ({ item }) => {
    const { name, calories, proteins, carbohydrates, fats } = item
    return (

        <View>
            <Text>{name}</Text>
            <Text> Calories: {calories} cal</Text>
            <Text> Protein: {proteins} g </Text>
            <Text> Carbohydrates: {proteins} g </Text>
            <Text> Fats: {proteins} g </Text>
        </View>
    )
}