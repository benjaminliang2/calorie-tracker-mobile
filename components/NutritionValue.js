import * as Progress from 'react-native-progress'
import { Text, View } from "react-native"


export const NutritionValue = (props) => {
    const { title, valueConsumed, valueRequired } = props


    let label = ''
    switch (title) {
        case 'Calories':
            label = 'cal'
            break
        default:
            label = 'g'
    }

    return (
        <View style={styles.macrocontainer}>
            <Text style={styles.title}> {title} </Text>
            <Progress.Bar progress={valueConsumed/valueRequired} width={75} height={2} color="#5F8D4E" />
            <Text style={styles.subtitle}> {valueConsumed} / {valueRequired} {label} </Text>
        </View>
    )

}

const styles = {
    macrocontainer: {
        alignItems: 'center',
        padding: 7,
        width: '100%',
        marginVertical: 5,


    },
    title: {
        fontSize: 15,
        fontFamily: 'MontserratSemiBold',
        color: '#A4BE7B'
    },
    subtitle: {
        fontSize: 15,
        fontFamily: 'MontserratBold',
        color: '#285430'

    }
}