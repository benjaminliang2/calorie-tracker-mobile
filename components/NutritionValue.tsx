import * as Progress from 'react-native-progress'
import { Text, View, StyleSheet } from "react-native"

interface Item {
    title: string,
    valueConsumed: number, 
    valueRequired: number,
}



export const NutritionValue = (props: Item): JSX.Element => {
    const { title, valueConsumed, valueRequired } = props


    let label:string = ''
    switch (title) {
        case 'Calories':
            label = 'cal'
            break
        default:
            label = 'g'
    }

    return (
        <View style={styles.macrocontainer}>
            <Text style={styles.title}> {title} </Text>
            <Progress.Bar progress={valueConsumed/valueRequired} width={75} height={2} color="#5F8D4E" />
            <Text style={styles.subtitle}> {valueConsumed} / {valueRequired} {label} </Text>
        </View>
    )

}

const styles = StyleSheet.create ({
    macrocontainer: {
        alignItems: 'center',
        padding: 7,
        width: '100%',
        marginVertical: 5,


    },
    title: {
        fontSize: 15,
        fontFamily: 'MontserratSemiBold',
        color: '#A4BE7B'
    },
    subtitle: {
        fontSize: 15,
        fontFamily: 'MontserratBold',
        color: '#285430'

    }
})