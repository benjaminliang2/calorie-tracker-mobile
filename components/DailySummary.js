import { useSelector } from "react-redux";
import { SafeAreaView, Keyboard, StyleSheet, Text, TouchableOpacity, TouchableWithoutFeedback, View } from "react-native"
import CircularProgress from "react-native-circular-progress-indicator";

import { NutritionValue } from './NutritionValue';


export const DailySummary = () => {
    const { calories, proteins, carbohydrates, fats } = useSelector((store) =>
        store.nutrition
    )
    return (<>
    <Text style={styles.title}>Today's Nutrition</Text>
        <View style={styles.container}>
            <View style={{  alignItems: 'center'}}>
                <CircularProgress
                    value={calories}
                    radius={80}
                    duration={2000}
                    progressValueColor={'#285430'}
                    maxValue={2000}
                    title={'cal'}
                    titleColor={'#285430'}
                    titleStyle={{ fontWeight: 'bold' }}
                />
            </View>
            <View style={{  alignItems: 'center' }} >
                <NutritionValue title={'Protein'} valueConsumed={proteins} valueRequired={150}/>
                <NutritionValue title={'Carb'} valueConsumed={carbohydrates} valueRequired={300} />
                <NutritionValue title={'Fat'} valueConsumed={fats} valueRequired={90}/>
            </View>

        </View>

    </>)
}

const styles ={
    title: {
        marginLeft: 10,
        marginTop: 15,
        marginBottom: 8,
        fontFamily: 'MontserratBold',
        fontSize: 28, 
    },
    container: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        paddingVertical: 20,
        margin: 10,
        borderRadius: 20,
        backgroundColor: '#BDF0CC',

    }
}