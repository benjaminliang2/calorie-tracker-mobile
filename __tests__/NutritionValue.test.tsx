import renderer from 'react-test-renderer';
import { NutritionValue } from '../src/components/NutritionValue';

it('renders correctly', () => {
    const tree = renderer.create(<NutritionValue title="testing" valueConsumed={10} valueRequired={100}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });