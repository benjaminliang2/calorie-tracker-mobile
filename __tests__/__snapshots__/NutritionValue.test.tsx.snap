// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginVertical": 5,
      "padding": 7,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#A4BE7B",
        "fontFamily": "MontserratSemiBold",
        "fontSize": 15,
      }
    }
  >
     
    testing
     
  </Text>
  <View
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    indeterminate={false}
    indeterminateAnimationDuration={1000}
    onLayout={[Function]}
    progress={0.1}
    style={
      Array [
        Object {
          "backgroundColor": undefined,
          "borderColor": "#5F8D4E",
          "borderRadius": 4,
          "borderWidth": 1,
          "overflow": "hidden",
          "width": 75,
        },
        undefined,
      ]
    }
    useNativeDriver={false}
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#5F8D4E",
          "height": 2,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateX": -32.85,
            },
            Object {
              "scaleX": 0.10009000000000001,
            },
          ],
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#285430",
        "fontFamily": "MontserratBold",
        "fontSize": 15,
      }
    }
  >
     
    10
     / 
    100
     
    g
     
  </Text>
</View>
`;
